src/component/axis/CartesianAxisView.js

```diff
@@ -109,7 +109,14 @@ var CartesianAxisView = AxisView.extend({
         var p2 = [];
 
         var lineStyle = lineStyleModel.getLineStyle();
+        // the showAsAxis will set lineColor to 'transparent' when the viewLabels value is undefind
+        var showAsAxis = splitLineModel.get('showAsAxis');
+        var viewLabels = showAsAxis && axis.getViewLabels();
+
         for (var i = 0; i < ticksCoords.length; i++) {
+            if(viewLabels && viewLabels[i] && !viewLabels[i].formattedLabel) {
+                continue;
+            }
```

Comment:
It seems that the changes in this file are related to #12487, which should not be here.